src/navigator/IFrameNavigator.ts

```diff
@@ -1799,65 +1894,106 @@ export default class IFrameNavigator implements Navigator {
       }
     } else {
       if (
-        (this.publication.metadata.rendition?.layout ?? "unknown") === "fixed"
+        (this.publication.Metadata.Rendition?.Layout ?? "unknown") === "fixed"
       ) {
-        if (even) {
-          if (isSameOrigin) {
-            this.iframe.src = this.currentChapterLink.href;
-            if (this.iframe2) {
-              if (index < this.publication.readingOrder.length - 1) {
-                const next = this.publication.getNextSpineItem(
-                  this.currentChapterLink.href
-                );
-                this.iframe2.src = this.publication.getAbsoluteHref(next.href);
-              } else {
-                this.iframe2.src = "about:blank";
+        if (this.settings.columnCount !== 1) {
+          if (even) {
+            if (isSameOrigin) {
+              this.iframes[0].src = this.currentChapterLink.href;
+              this.currentSpreadLinks.left = {
+                href: this.currentChapterLink.href,
+              };
+
+              if (this.iframes.length == 2) {
+                if (index < this.publication.readingOrder.length - 1) {
+                  const next = this.publication.getNextSpineItem(
+                    this.currentChapterLink.href
+                  );
+                  var href = this.publication.getAbsoluteHref(next.Href);
+                  this.iframes[1].src = href;
+                  this.currentSpreadLinks.right = {
+                    href: href,
+                  };
+                } else {
+                  this.iframes[1].src = "about:blank";
+                }
+              }
+            } else {
+              fetch(this.currentChapterLink.href)
+                .then((r) => r.text())
+                .then(async (content) => {
+                  writeIframeDoc.call(
+                    this,
+                    content,
+                    this.currentChapterLink.href
+                  );
+                });
```

Comment:
Doesn't need to be changed here, but just so you know, this can be rewritten with async/await as:
```ts
const r = await fetch(this.currentChapterLink.href)
const content = await r.text();
writeIframeDoc.call(this, content, this.currentChapterLink.href)
```
which avoids then nesting of `.then` calls. Basically anything you would call `something().then(value => {...})` you can use `const value = await something()` so long as you're in an `async` function

Reply:
but looks much better :) thank you, will change that

Reply:
actually i will leave it for now as is, since it seems to require more changes to make it work.